ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobsSidecarAvailabilityChecker.java

```diff
@@ -0,0 +1,106 @@
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.SpecVersion;
+import tech.pegasys.teku.spec.datastructures.blocks.BeaconBlock;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.BeaconBlockBody;
+import tech.pegasys.teku.spec.datastructures.blocks.blockbody.versions.eip4844.BeaconBlockBodyEip4844;
+import tech.pegasys.teku.spec.datastructures.execution.versions.eip4844.BlobsSidecar;
+import tech.pegasys.teku.spec.datastructures.type.SszKZGCommitment;
+import tech.pegasys.teku.spec.logic.versions.eip4844.blobs.BlobsSidecarAvailabilityChecker;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class ForkChoiceBlobsSidecarAvailabilityChecker implements BlobsSidecarAvailabilityChecker {
+  private SpecVersion specVersion;
+  private RecentChainData recentChainData;
+  private SignedBeaconBlock block;
+  private Optional<BlobsSidecar> blobsSidecar;
+
+  public ForkChoiceBlobsSidecarAvailabilityChecker(
+      final SpecVersion specVersion,
+      final RecentChainData recentChainData,
+      final SignedBeaconBlock block,
+      final Optional<BlobsSidecar> blobsSidecar) {
+    this.specVersion = specVersion;
+    this.recentChainData = recentChainData;
+    this.block = block;
+    this.blobsSidecar = blobsSidecar;
+  }
+
+  @Override
+  public boolean retrieveBlobsSidecar() {
+    return true;
+  }
+
+  @Override
+  public SafeFuture<BlobsSidecarAndValidationResult> validateBlobsSidecar() {
+
+    // in the current 4844 specs, the blobsSidecar is immediately available with the block
+    // so if we have it we do want to validate them regardless
+    if (blobsSidecar.isPresent()) {
+      return validate(blobsSidecar.get());
+    }
+
+    if (isBlockInDataAvailabilityWindow()) {
+      return BlobsSidecarAvailabilityChecker.NOT_AVAILABLE_RESULT;
+    }
+
+    return BlobsSidecarAvailabilityChecker.NOT_REQUIRED_RESULT;
+  }
+
+  private SafeFuture<BlobsSidecarAndValidationResult> validate(final BlobsSidecar blobsSidecar) {
+    final BeaconBlockBodyEip4844 blockBody =
+        block
+            .getBeaconBlock()
+            .map(BeaconBlock::getBody)
+            .flatMap(BeaconBlockBody::toVersionEip4844)
+            .orElseThrow();
+
+    if (!specVersion
+        .miscHelpers()
+        .isDataAvailable(
+            block.getSlot(),
+            block.getBodyRoot(),
+            blockBody.getBlobKzgCommitments().stream()
+                .map(SszKZGCommitment::getKZGCommitment)
+                .collect(Collectors.toUnmodifiableList()),
+            blobsSidecar)) {
+      return BlobsSidecarAvailabilityChecker.invalidResult(blobsSidecar);
+    }
+
+    return BlobsSidecarAvailabilityChecker.validResult(blobsSidecar);
+  }
+
+  private boolean isBlockInDataAvailabilityWindow() {
+    final UInt64 currentSlot = recentChainData.getCurrentSlot().orElseThrow();
+
+    if (block
```

Comment:
If I'm not wrong, it will be always `false`. Also it could be simplified. And I'd add test for it

Reply:
give another look now. test added